coremltools/models/_compiled_model.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2023, Apple Inc. All rights reserved.
+#
+# Use of this source code is governed by a BSD-3-clause license that can be
+# found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+
+from os.path import expanduser as _expanduser
+
+from .model import MLModel as _MLModel
+from .utils import _macos_version
+from coremltools import ComputeUnit as _ComputeUnit
+
+try:
+    from ..libcoremlpython import _MLModelProxy
+except:
+    _MLModelProxy = None
+
+
+class CompiledMLModel:
+    def __init__(self, path: str, compute_units: _ComputeUnit =_ComputeUnit.ALL):
+        """
+        Loads a compiled Core ML model.
+
+        Parameters
+        ----------
+        path : str
+            The path to a compiled model directory, ending in ".mlmodelc".
+
+        compute_units : coremltools.ComputeUnit
+
+            An enum with the following possible values.
+
+                - ``coremltools.ComputeUnit.ALL``: Use all compute units available, including the
+                 neural engine.
+                - ``coremltools.ComputeUnit.CPU_ONLY``: Limit the model to only use the CPU.
+                - ``coremltools.ComputeUnit.CPU_AND_GPU``: Use both the CPU and GPU, but not the
+                 neural engine.
+                - ``coremltools.ComputeUnit.CPU_AND_NE``: Use both the CPU and neural engine, but
+                  not the GPU. Available only for macOS >= 13.0.
+
+        Examples
+        --------
+        .. sourcecode:: python
+
+        >>> my_compiled_model = ct.models.CompiledMLModel("my_model_path.mlmodelc")
+        >>> y = my_compiled_model.predict({'x': 3})
+
+        See Also
+        --------
+        predict
+        """
+        if _macos_version() < (10, 13):
+            raise Exception("Loading compiled Core ML models is only support on macOS 10.13 or higher.")
+        if _MLModelProxy is None:
+            raise Exception("Unable to load any compiled models.")
```

Comment:
Maybe add a few more details here? Maybe that either Core ML framework is not available (likely due to non macOS platform) or error loading Core ML.

Reply:
The non macOS platform case is handled by the previous if-statement. The most likely reason this exception will get triggered is if coremltools was install from an egg rather than a wheel. I'll add that info to the error message.